docs/reference-architectures/r.rst

```diff
@@ -0,0 +1,349 @@
+
+.. code-block:: sql
+
+    SELECT COUNT(*) FROM iris;
+    +----------+
+    | count(*) |
+    +----------+
+    |      150 |
+    +----------+
+    SELECT 1 row in set (0.130 sec)
+
+Examining The Data
+------------------
+
+With our data in CrateDB, we can now load it into R or RStudio. First, within R, we should
+import our data by loading the ``RPostgreSQL`` library, connecting to a
+CrateDB database and loading the dataset:
+
+.. code-block:: r
+
+    # Loading the DBI library.
+    > library(DBI)
+
+    # Opening a connection, where dbname is the name of our schema, host is the address
+    # of the CrateDB instance, port is the CrateDB Psql port, and user is the default
+    # crate user.
+    > con <- dbConnect(RPostgres::Postgres(),
+                       dbname = "doc",
+                       host = "localhost",
+                       port = 5432,
+                       user = "crate")
+
+    # Loading the iris dataset from CrateDB
+    > iris_dataset <- dbGetQuery(con, "SELECT * FROM doc.iris")
+
+We can then validate that we have loaded the dataset correctly by looking at the
+dimensions and the summary of the dataset:
+
+.. code-block:: r
+
+    # Getting the dimensions of the dataset.
+    > dim(iris_dataset)
+    [1] 150   5
+
+    # This indicates it contains 150 instances across 5 attributes. We can check
+    # the types of those attributes as well.
+    > sapply(iris_dataset, class)
+    sepal_length  sepal_width petal_length  petal_width      species
+    "numeric"    "numeric"    "numeric"    "numeric"  "character"
+
+    # Finally, we can look at the the statistical summary of our dataset.
+    > summary(iris_dataset)
+    sepal_length    sepal_width     petal_length    petal_width
+    Min.   :4.300   Min.   :2.000   Min.   :1.000   Min.   :0.100
+    1st Qu.:5.100   1st Qu.:2.800   1st Qu.:1.600   1st Qu.:0.300
+    Median :5.800   Median :3.000   Median :4.350   Median :1.300
+    Mean   :5.843   Mean   :3.054   Mean   :3.759   Mean   :1.199
+    3rd Qu.:6.400   3rd Qu.:3.300   3rd Qu.:5.100   3rd Qu.:1.800
+    Max.   :7.900   Max.   :4.400   Max.   :6.900   Max.   :2.500
+    species
+    Length:150
+    Class :character
+    Mode  :character
+
+We can now visualise our data. For example, visualising a boxplot of iris properties
+by each iris species might give us an insight about the distribution of these properties
+across each species.
+
+.. code-block:: r
+
+    # Boxplot of Iris features by each species
+    > par(mfrow=c(2,2))
+    > boxplot(sepal_length~species, data=iris_dataset, main="Sepal Length by Species")
+    > boxplot(sepal_width~species, data=iris_dataset, main="Sepal Width by Species")
+    > boxplot(petal_length~species, data=iris_dataset, main="Petal Length by Species")
+    > boxplot(petal_width~species, data=iris_dataset, main="Petal Width by Species")
+
+.. figure:: r-iris-features.png
+   :align: center
+
+As we can see, sepal length/width and petal length/width are very good indicators of
```

Comment:
Wondering whether the solidus here falls under the exceptions: https://intranet.crate.io/handbook/conventions/docs/style.html#miscellaneous